File: 2429_minimize_xor.cpp
class Solution {
public:
    int minimizeXor(int z, int y) {
        int need = __builtin_popcount(y), have = __builtin_popcount(z);
        while (need != have) {
            z += (have < need ? 1 : -1);
            have = __builtin_popcount(z);
        }
        return z;
    }
};

File: 2429_minimize_xor.py
class Solution(object):
    def minimizeXor(self, m, n):
        req = bin(n).count('1')
        res = 0
        for i in range(31, -1, -1):
            if req and (m & (1 << i)):
                res |= (1 << i)
                req -= 1
        for i in range(32):
            if req and not (res & (1 << i)):
                res |= (1 << i)
                req -= 1
        return res
